src/network.rs

```diff
@@ -32,8 +32,14 @@ use std::fmt::Debug;
 pub trait Network<H: Hasher, D: Data, S: Signature, MS: PartialMultisignature>: Send {
     type Error: Debug;
     /// Send a message to a single node.
+    ///
+    /// Note on the implementation: this function should be implemented in a non-blocking manner.
+    /// Otherwise, the performance might be affected negatively.
```

Comment:
I would say sth stronger like "Otherwise, the performance might be affected negatively or the execution may end up in a dead lock" or just "Otherwise, the execution of the consensus may end up in a dead lock".

Reply:
done